presto-main/src/main/java/com/facebook/presto/util/JsonUtil.java

```diff
@@ -239,6 +250,12 @@ static ObjectKeyProvider createObjectKeyProvider(Type type)
                 case StandardTypes.VARCHAR:
                     return (block, position) -> type.getSlice(block, position).toStringUtf8();
                 default:
+                    if (type instanceof LongEnumType) {
```

Comment:
Conceptually this should not be in default... `TypeSignature` for enum types should work properly.